### ISSUE TITLE
Wrong answer for ask(Q.rational(x)) for a float x

### ISSUE THREAD

**Issue by meet7202** [sympy/sympy#12168]
Wrong answer for `ask(Q.rational(x))` for `x` a float instance
e.g
```
>>> ask(Q.rational(math.pi))
True  # this should return None
```
I want to solve this Issue can someone guide me.

**Comment by jksuom** [comment]
`math.pi` is internally converted to a Float that will be different from `pi` even if it is a good approximation. Hence it cannot be expected to have the properties of `pi`.

However, the last result (`True`) is wrong. To fix it this [Float handler](https://github.com/sympy/sympy/blob/master/sympy/assumptions/handlers/sets.py#L135) of rationality should be replaced by one always returning None.

**Comment by gxyd** [comment]
@jksuom does `Q.rational(x)` checks whether `x` is mathematically a rational number or not? If it does check that, then isn't `math.pi` actually a rational number. So isn't actually the last answer correct?

**Comment by asmeurer** [comment]
I think we at some point decided that the assumptions returning None for Float.is_rational is the most reasonable thing. The new assumptions should do the same.

**Comment by jksuom** [comment]
A Float is an approximate representation of a real number. The (potential) number of Floats is countably infinite while the cardinality of the set of real numbers is bigger ("the cardinality of continuum"). Hence Floats cannot correspond to real numbers in a one-to-one way. Instead, each Float corresponds to an infinite set of real numbers, both rational and irrational. Therefore there can be no definite value for `Q.rational(x)` when `x` is a Float.

**Comment by asmeurer** [comment]
The way I think about it is that floats don't behave like rationals with respect to operations, so you can run into trouble with the assumptions which logically assume rationals have certain properties. For instance, floats don't go to zero when they are supposed to (even a + b - a - b could be nonzero).

**Comment by gxyd** [comment]
Both of your comments have helped me to create a good distinction between the Float's and Rational's. Thinking along the Kalevi's comment direction really clarified.

I've marked the issue as easy to fix.

**Comment by ArighnaIITG** [comment]
@gxyd , @jksuom, 
I would like to work on it.
Can you please guide me as to how I should proceed?

**Comment by gxyd** [comment]
Thanks for your interest, I think someone already asked on gitter, I think @abhigyank. So may be we should wait, if he submits a PR.

**Comment by abhigyank** [comment]
Yes I am working on it.

**Comment by abhigyank** [comment]
@gxyd From the discussion above, what changes are to be made? I am a bit unclear on the math.

**Comment by gxyd** [comment]
This assignment https://github.com/sympy/sympy/blob/master/sympy/assumptions/handlers/sets.py#L135 assigns both `Rational` and `Float` to `staticmethod(CommonHandler.AlwaysTrue)`.
See:
```python
>>> a, b = [2]*2
>>> a
2
>>> b
2
```
Here we'll need a function decorated with `@staticmethod` returning `None` for `Float`.

**Comment by abhigyank** [comment]
I am getting an issue here. If I change``` Float``` to ```staticmethod(CommonHandler.Float)``` where ```CommonHandler.Float``` returns ```None```, the output for ask(Q.rational(1.4)) would also return ```None`` . Thus it fails the existing float tests. What should be done?

**Comment by gxyd** [comment]
Modify the tests as well, I guess.

**Comment by gxyd** [comment]
Closing the issue, since fixed by #13870 .
